src/runtime/bitmap.h

```diff
@@ -42,7 +42,8 @@ static inline boolean bitmap_extend(bitmap b, u64 i)
 {
     if (i >= b->mapbits) {
 	b->mapbits = pad(i + 1, ALLOC_EXTEND_BITS);
-	extend_total(b->alloc_map, b->mapbits >> 3);
+	if (!extend_total(b->alloc_map, b->mapbits >> 3));
```

Comment:
Not sure why, but I get compiler errors when I move line 44 (setting b->mapbits) below extend_total, so I kept it as-is.

Reply:
I don't know what compiler errors you got, but I'm sure they can be solved...

Reply:
@francescolavra would you have any insight as to why it would be segfaulting? Here's the last few lines of the out from running ```make```:
```
MKFS	image
UUID: ab742659-1b68-aa39-218d-a9aa0d615f95
metadata (children:(kernel:()))
Segmentation fault (core dumped)
make[1]: *** [Makefile:236: image] Error 139
make[1]: Leaving directory '/home/esther/virtualization/nanos/platform/pc'
make: *** [Makefile:48: image] Error 2
```

Reply:
I don't know why it's segfaulting, I tried your branch and here `make` runs just fine, you may want to check any local modifications you may have done.
The segfault is happening in the mkfs tool, whose source is at tools/mkfs.c. When running `make`, this tool is invoked as
`./output/tools/bin/mkfs  -b output/platform/pc/boot/boot.img -k output/platform/pc/bin/kernel.img output/image/disk.raw < test/runtime/webg.manifest`

Reply:
Does it segfault for you after moving the `b->mapbits` update below the `extend_total `check? That's what I was referring to.

I believe I've tracked down the problem, but am unsure of how to fix it. Upon gdbing mkfs, it showed the segfault occurring in src/runtime/bitmap.c, line 131, which dereferences the result of a function that uses the `mapbase` variable. `mapbase` is only updated if `bitmap_extend` succeeds, so the update to `mapbits` (moving line 44 above below the `extend_total` check) only happens upon allocation success. Returning `INVALID_PHYSICAL` if `bitmap_extend` on bitmap.c, line 127 fixes the segfault problem, but causes MKFS to fail due to an alloc fail. Not sure if this indicates that line 44 shouldn't be moved, that the conditional causing the segfault should be removed, or something else?

Reply:
Just moving the `b->mapbits = pad(i + 1, ALLOC_EXTEND_BITS)` line below the extend_total() call is not correct, because the extend_total() call uses `b->mapbits`, which after moving the above line is no longer what it used to be. The correct way to implement this is by using a local variable, as in:
```
    u64 mapbits = pad(i + 1, ALLOC_EXTEND_BITS);
    if (extend_total(b->alloc_map, mapbits >> 3)) {
        b->mapbits = mapbits;
        return true;
    }
```

Reply:
Right, that makes sense, thank you. I've made the other changes you requested as well.